error.go

```diff
@@ -227,11 +227,9 @@ func (e *Error) AddDetail(d *ErrorDetail) {
 
 // Meta allows the error to carry additional information as key-value pairs.
 //
-// Metadata attached to errors returned by unary handlers is always sent as
-// HTTP headers, regardless of the protocol. Metadata attached to errors
-// returned by streaming handlers may be sent as HTTP headers, HTTP trailers,
-// or a block of in-body metadata, depending on the protocol in use and whether
-// or not the handler has already written messages to the stream.
+// Metadata attached to errors is returned either as HTTP headers or HTTP trailers
+// depending on the protocol, whether it is unary or streaming, and whether or
+// not the handler has already written messages to the stream.
```

Comment:
I think we could be more accurate here and state that metadata attached to a handler error is accessible via Error.Metadata()` in the client, but it is also accessible via `CallInfo.ResponseTrailers` regardless of protocol.

I don't think it makes any sense to describe things about the physical transport, since that is protocol specific. Even though you widened it to allow it in HTTP headers or HTTP trailers, that isn't even accurate since streaming operations in gRPC-Web and Connect actually send them in the body.

Reply:
But connect-go always makes them available as the `ResponseHeader` and `ResponseTrailer` and they are _not_ currently always available in the `CallInfo.ResponseTrailers` (it depends on the protocol).

Reply:
> (it depends on the protocol)

This seems like a serious bug to me. Why does the caller have to be cognizant of the protocol? The point of multi-protocol support in this library is to provide consistent semantics, regardless of protocol.

Reply:
Yeah, I'm convinced that you're right.

For Go and ES, previous code that uses `Error.Meta()` should be fine. But changing this so that Connect Unary error metadata is (semantically) in trailers instead of headers would be visible and arguably a breaking change for Kotlin and Swift, that surface metadata differently.

Reply:
With gRPC's "trailers-only" response, an RPC that produces an immediate error _may_ respond with a single terminal HEADER frame. The server might provide an API that allows setting header fields and error metadata or trailer fields separately, but the trailers-only response does not distinguish between them on the wire. In gRPC-Web, this translates to a HTTP 1.1 response with headers, and without a body. 

Because of this property of the gRPC protocol, it isn't possible to provide consistent behavior in an API for errors that exposes headers and trailers separately.